src/engine/Operation.h

```diff
@@ -282,11 +282,13 @@ class Operation {
   // its result.
   [[nodiscard]] virtual bool supportsLimit() const { return false; }
 
+  // Hook subclasses can override to pass changes to the limit to their children
+  virtual void onLimitChanged(const LimitOffsetClause&) const {}
```

Comment:
"This function is called each time `applyLimit` is called. It can be overridden by subclasses to e.g. implement the LIMIT in a more efficient way".

And this function should probably be protected or even private, shouldn't it?